modules/social_features/social_group/modules/social_group_default_route/src/EventSubscriber/RedirectSubscriber.php

```diff
@@ -100,13 +103,56 @@ public function groupLandingPage(RequestEvent $event) {
       return;
     }
 
+    $this->doRedirect($event, $group);
+  }
+
+  /**
+   * Redirect on exceptions.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\ExceptionEvent $event
+   *   The exception event.
+   */
+  public function onKernelException(ExceptionEvent $event): void {
+    $exception = $event->getThrowable();
```

Comment:
I added it because we will get a bit when non-member clicks "Request to join" to the group, he will be redirected to "stream" by default - but it is still forbidden for non-members. I think we could refactor it in the scope of global refactoring of redirection, but it is a important hotfix to don't broke anything